sdk/core/Azure.Core/src/DefaultClientOptions.cs

```diff
@@ -10,7 +10,6 @@ internal class DefaultClientOptions: ClientOptions
     {
         public DefaultClientOptions(): base(null, null)
         {
-            Transport = HttpPipelineTransport.Create();
```

Comment:
Was this redundant?

Reply:
setting of the public property (rather than the private backing field indicates to the pipeline that the Transport has been replaced by the customer and therefor TransportOptions cannot be safely applied.

Reply:
It's done here also https://github.com/Azure/azure-sdk-for-net/pull/28645/files/21d226fbef097e370ae4a2d3629991407a124096#diff-e39b05dd3e9930120929e0513ea6029c1e4e44372c7bb0d896d6260b4afbedb0R64

Reply:
I see.. so is this actually a bug fix?

Reply:
Yes - although it currently only affects clients that set TransportOptions.